src/libleo/leo/leofunc.c

```diff
@@ -0,0 +1,135 @@
+#include "global.h"
+#include "ultra64/leo.h"
+#include "ultra64/leoappli.h"
+#include "ultra64/leodrive.h"
+
+static s32 __leoResetCalled = false;
+static s32 __leoQueuesCreated = false;
+
+static OSMesgQueue LEOpost_que;
+static OSMesg LEOpost_que_buf[1];
+
+void leoInitialize(OSPri compri, OSPri intpri, OSMesg* command_que_buf, u32 cmd_buff_size) {
+    u32 savedMask;
+    OSPri oldPri;
+    OSPri myPri;
+    OSPri pri;
+
+    if (intpri < compri) {
+        pri = compri;
+    } else {
+        pri = intpri;
+    }
+
+    oldPri = -1;
+
+    myPri = osGetThreadPri(NULL);
+    if (myPri < pri) {
+        oldPri = myPri;
+        osSetThreadPri(NULL, pri);
+    }
+
+    savedMask = __osDisableInt();
+
+    __leoQueuesCreated = true;
+    osCreateMesgQueue(&LEOcommand_que, command_que_buf, cmd_buff_size);
+    osCreateMesgQueue(&LEOcontrol_que, LEOcontrol_que_buf, ARRAY_COUNT(LEOcontrol_que_buf));
+    osCreateMesgQueue(&LEOevent_que, LEOevent_que_buf, ARRAY_COUNT(LEOevent_que_buf));
+    osCreateMesgQueue(&LEOdma_que, LEOdma_que_buf, ARRAY_COUNT(LEOdma_que_buf));
+    osCreateMesgQueue(&LEOblock_que, LEOblock_que_buf, ARRAY_COUNT(LEOblock_que_buf));
+    osCreateMesgQueue(&LEOpost_que, LEOpost_que_buf, ARRAY_COUNT(LEOpost_que_buf));
+    osCreateThread(&LEOcommandThread, 1, leomain, NULL, STACK_TOP(LEOcommandThreadStack), compri);
+    osStartThread(&LEOcommandThread);
+    osCreateThread(&LEOinterruptThread, 1, leointerrupt, NULL, STACK_TOP(LEOinterruptThreadStack), intpri);
+    osStartThread(&LEOinterruptThread);
+    osSetEventMesg(2, &LEOevent_que, (OSMesg)0x30000);
+    osSendMesg(&LEOblock_que, NULL, OS_MESG_NOBLOCK);
+    __osRestoreInt(savedMask);
+
+    if (oldPri != -1) {
+        osSetThreadPri(NULL, oldPri);
+    }
+}
+
+void leoCommand(void* cmd_blk_addr) {
+    if (__leoResetCalled != 0) {
+        ((LEOCmd*)cmd_blk_addr)->header.status = LEO_STATUS_CHECK_CONDITION;
+        ((LEOCmd*)cmd_blk_addr)->header.sense = LEO_SENSE_WAITING_NMI;
+        if ((((LEOCmd*)cmd_blk_addr)->header.control & LEO_CONTROL_POST) != 0) {
+            osSendMesg(((LEOCmd*)cmd_blk_addr)->header.post, (OSMesg)LEO_SENSE_WAITING_NMI,
+                       OS_MESG_BLOCK); // Presumably
```

Comment:
What does the comment here mean?

Reply:
I don't know, I think it refers to `LEO_SENSE_WAITING_NMI` for `37` here. Do you think it's correct?

Reply:
From a quick look it seems to make sense to use the sense defines as the messages. I'd remove the comment or elaborate on it, right now it's just confusing.